data/features/m/m.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Description: Function to search arguments in the manual of a command
+# Argument 1: String representing a system binary (example: ls)
+# Argument 2 (optional): String representing argument option of the binary (example: -A)
+m() {
+  if ! command -v man &>/dev/null; then
+    echo "Error: 'man' command not found."
+    return 1
+  fi
+
+  if [ $# -eq 0 ]; then # No arguments given
+    echo "Error: Invalid arguments."
+    return 1
+  fi
+
+  if [ $# -eq 1 ]; then
+    # Matches any line that starts with one or more spaces followed by a dash (-), one letter (lowercase or uppercase),
+    # a comma, a space, two hyphens (--), and one or more letters (lowercase or uppercase) or dashes (-).
+    # This pattern is used to match the short and long options that are documented in the manual page.
+    man_output="$(man "$@" | grep -E '^ +-[a-zA-Z], --[a-zA-Z-]+')"
+    echo "${man_output}"
+  else
+    for arg in "${@:2}"; do
+      # search for lines that start with the second argument ($2) preceded by one or more spaces (^ +) and followed by
+      # a word boundary (\b). The -A5 option tells grep to also output five lines of context after each match.
+      # Remove the leading whitespace from the lines that were matched in the previous step.
+      man "$1" | grep -E "^ +${arg}\\b" -B 5 -A 5
```

Comment:
same here with regexp

Reply:
Don't know how to solve this.

Reply:
Only search for the argument, delete the rest of matching format

Reply:
<code>man "$1" | grep -hnI -E "^ +${arg}" -B 5 -A 5 --color='auto'</code>

Reply:
Yes, better, but only search the argument!

Reply:
If only the argument, it shows other commands. We only want to show the one that interest us ($2)

Reply:
So this should be it:
<code>man "$1" | grep -hnI -E "+${arg}" -B 5 -A 5 --color='auto'</code>

Reply:
solved